src/geometry/LinearPropagator.i.hh

```diff
@@ -46,16 +50,20 @@ void LinearPropagator::operator()()
  * \pre Assumes that next_step() has been properly called by client
  */
 CELER_FUNCTION
-void LinearPropagator::operator()(real_type dist)
+LinearPropagator::OutputType LinearPropagator::operator()(real_type dist)
 {
     REQUIRE(dist > 0.);
-    REQUIRE(dist <= track_.next_step());
-    this->apply_linear_step(dist);
+    OutputType result;
+    const real_type minstep = fmin(dist, track_.next_step() + track_.tolerance());
+    result.distance = this->apply_linear_step( minstep );
 
-    if (std::fabs(track_.next_step()) < track_.tolerance())
-    {
+    //.. for linear trajectory, assume that next_step() takes it to boundary
+    if (track_.next_step() < track_.tolerance()) {
```

Comment:
There is no 'extra' conditional.  The conditional is required so that move_volume() only gets called when dist > next_step_.  Well, I will try to move some things around...